### PR TITLE
Crop to preview

### PR DESCRIPTION
**Pull request by ejkkan**
The cropping of passing the "preview" parameter to captureQuality did not work as expected. 
Since one should still have the choice of setting a certain image quality when wanting the image cropped aswell, a new property has been added so that one can distinguish between the two.

Property = cropToPreview={true} // default false

If true, the image will be cropped to the exact ratio of the camera preview window for both Android and IOS.

should be fixing some issues like: #765  #37  #648

### CODE REVIEW

**Comment by mckn** [comment]
@lwansbrough is there anything more we can do to get this PR accepted? I will update the docs, but appart from that?

**Comment by ejkkan** [comment]
@sibelius Can anything more be done to have this PR merged? We greatly need this functionality. 
Thank you

File: ios/RCTCameraManager.h [hunk 1]
```diff
@@ -72,11 +71,11 @@ typedef NS_ENUM(NSInteger, RCTCameraTorchMode) {
 @property (nonatomic, assign) NSInteger videoTarget;
 @property (nonatomic, assign) NSInteger orientation;
 @property (nonatomic, assign) BOOL mirrorImage;
+@property (nonatomic, assign) BOOL cropToPreview;
 @property (nonatomic, strong) NSArray* barCodeTypes;
 @property (nonatomic, strong) RCTPromiseResolveBlock videoResolve;
 @property (nonatomic, strong) RCTPromiseRejectBlock videoReject;
 @property (nonatomic, strong) RCTCamera *camera;
-@property (nonatomic, assign) BOOL cropToViewport;
```
Comment (sibelius) [inline ios/RCTCameraManager.h]: is this a breaking change ?

Comment (mckn) [reply ios/RCTCameraManager.h]: Well, it is compared to the latest in master. But I don't think the version with that cropToViewport is released as a package?

We replaced the cropToViewport with cropToPreview and also implemented it so it works for all orientations for both android and iOS.

The old implementation was merged into master in this commit:
https://github.com/react-native-community/react-native-camera/commit/39f19cd7c7b39689f32af7ad08a26e248f0742f9

@sibelius

Comment (sibelius) [reply ios/RCTCameraManager.h]: well, it is time for a new release

@lwansbrough can we help us with that?

@mckn fix the conflicts and let's merge this, we can test more on master

**Comment by sibelius** [comment]
@Ignigena what do you think of this?

**Comment by Ignigena** [comment]
@sibelius the functionality looks to be the same judging by the code itself (don't have time to spin it up locally right now)

Definitely all for consistency between Android and iOS implementations if the change of naming gets us there. The previous was used since iOS doesn't have a "preview" capture quality mode and to try to replicate the existing Android preset name and function as much as possible.

**Comment by mckn** [comment]
Well, that is one of the things that we did not really understand. Did we have a capture quality that was preview? If so how did that work? This feature crops the captured image to what you see inside the preview. 

The image still keeps the previous selected capture image quality.

Example: I can capture an image with high quality and still crop it to a square if we only display a square to our end user in the preview.

**Comment by Ignigena** [comment]
@mckn As I understood it, previously the `preview` option on Android would automatically set the device capture quality to the closest match to the viewport. My changes on iOS only brought software cropping to match the viewport but left the device capture quality as high without attempting to match the resolution of the viewport on capture.

Seems with your changes in this PR things are a bit more consistent between platforms which is definitely 👍

**Comment by sibelius** [comment]
great work guys

tks

**Comment by sunweiyang** [comment]
Thanks for this! However I'm experiencing an issue on the latest master, which includes this commit. 

On iOS it works great, but on Android I'm seeing that it only crops the left/right sides off. In other words, I've defined my preview to be a square, but what I get instead is a tall skinny rectangular captured image.

Anyone else seeing this?

**Comment by EskelCz** [comment]
Doesn't work for me, on Xiaomi Redmi 6A. The preview is not even respecting the width and height in style, on RNCamera component.

**Pull request merged**